table/server/common/src/main/java/alluxio/table/common/udb/UdbFilterSpec.java

```diff
@@ -0,0 +1,505 @@
+    }
+    Mode partitionMode = mBypassedPartitions.get(tableName).getFirst();
+    boolean isBypassed =
+        matchedByTemplates(mBypassedPartitions.get(tableName).getSecond(), partitionName);
+    return shadowedByIgnore(tableName, handleInclusionExclusion(partitionMode, isBypassed));
+  }
+
+  private static boolean handleInclusionExclusion(Mode mode, boolean selected) {
+    switch (mode) {
+      case INCLUDE:
+        return selected;
+      case EXCLUDE:
+        return !selected;
+      case NONE:
+      default:
+        return false;
+    }
+  }
+
+  /**
+   * Checks if a table is shadowed by ignoring when it is also bypassed.
+   * @param tableName the table name
+   * @param mightBeShadowedIfTrue the return value of {@link #isBypassedTable(String)}, etc
+   * @return false if the table is shadowed by ignoring, mightBeShadowedIfTrue if it is not
+   */
+  private boolean shadowedByIgnore(String tableName, boolean mightBeShadowedIfTrue) {
+    if (mightBeShadowedIfTrue && isIgnoredTable(tableName)) {
+      // todo(bowen): this warning is never triggered bc if a table is ignored,
+      // it will not be checked for bypassing, given the current implementation in Database
+      LOG.warn("Table {} is set to be bypassed but it is also ignored", tableName);
+      return false;
+    }
+    return mightBeShadowedIfTrue;
+  }
+
+  private static boolean matchedByTemplates(Set<NameTemplate> templates, String name) {
+    return templates.stream().anyMatch(p -> p.matches(name));
+  }
+
+  /**
+   * Abstract name for tables and partitions.
+   */
+  interface NameTemplate {
+    /**
+     * Checks whether the argument name matches the configured name pattern.
+     *
+     * @param name the concrete name to check for
+     * @return true if the names match, false otherwise
+     */
+    boolean matches(String name);
+  }
+
+  static class ExactNameTemplate implements NameTemplate {
+    private final String mExactName;
+
+    public ExactNameTemplate(String exactName) {
+      Preconditions.checkArgument(!exactName.isEmpty(), "empty name");
+      mExactName = exactName;
+    }
+
+    @Override
+    public boolean matches(String name) {
+      return mExactName.equals(name);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      ExactNameTemplate that = (ExactNameTemplate) o;
+      return Objects.equals(mExactName, that.mExactName);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(mExactName);
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("exactName", mExactName)
+          .toString();
+    }
+  }
+
+  static class PatternNameTemplate implements NameTemplate {
+    private final Pattern mPattern;
+
+    public PatternNameTemplate(Pattern pattern) {
+      mPattern = pattern;
```

Comment:
```suggestion
      mPattern = Preconditions.checkNotNull(pattern, "pattern");
```

Reply:
Fixed in 4c0380b7c97d564304be200db2f40eb680da9f84